maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt

```diff
@@ -179,7 +183,29 @@ class ApiClient(
 
         response.use {
             if (!response.isSuccessful) {
-                throw CliError("Upload request failed (${response.code}): ${response.body?.string()}")
+                if (response.code >= 500 && completedRetries < maxRetryCount) {
+                    PrintUtils.message("Request failed, retrying...")
```

Comment:
@dmitry-zaitsev the progress bar is present during the retries and is reset upon the next request, so this message is not showing up when testing locally. Do you know of a neat way to get this info out to the user? Worst case we can drop it as the progress bar resetting might a good enough of an indicator that it is being retired. Let me know what you think!

Reply:
I think that is fine, the message will show up in CI output anyway (as it does not erase the progress bar there)

Reply:
Great, thanks! 👍